src/org/zaproxy/zap/extension/ascanrulesBeta/UsernameEnumeration.java

```diff
@@ -139,11 +139,18 @@ public void scan() {
 		try {
 			boolean loginUrl = false;
 
-			// Are we dealing with a login url in any of the contexts of which this uri is part
-			URI requestUri = getBaseMsg().getRequestHeader().getURI();
 			ExtensionAuthentication extAuth = (ExtensionAuthentication) Control.getSingleton()
 					.getExtensionLoader().getExtension(ExtensionAuthentication.NAME);
 
+			if(extAuth.getModel().getSession().getContexts().isEmpty()) {
```

Comment:
Hmmmm that's strange this worked in my testing. Perhaps getContexts() doesn't return the default context?